### PR TITLE
[graphql client] add sub selection support

### PR DESCRIPTION
**Pull request by alangenfeld**
## Test Plan

added tests

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6397** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/6397" target="_blank">Graphite</a>  (this PR)
    * **PR #6398** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/6398" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

**Review by alangenfeld — COMMENTED** [review]
currently I have `op_selection` / `solid_selection`

I could avoid the dupe names by doing `subset_selection` as the arg name

File: python_modules/dagster-graphql/dagster_graphql_tests/client_tests/test_submit_pipeline_execution.py [hunk 1]
```diff
@@ -73,6 +76,32 @@ def test_job_tags_success(mock_client: MockClient):
...
 
 
+@python_client_test_suite
+def test_pipeline_subset_success(mock_client: MockClient):
+    mock_client.mock_gql_client.execute.return_value = launch_pipeline_success_response
+    actual_run_id = mock_client.python_client.submit_pipeline_execution(
+        "bar",
+        repository_location_name="baz",
+        repository_name="quuz",
+        solid_selection=[""],
```
Comment (yuhan) [inline python_modules/dagster-graphql/dagster_graphql_tests/client_tests/test_submit_pipeline_execution.py:86]: can we test some actual selection?

**Review by yuhan — COMMENTED** [review]
i guess solidSelection is fine because we also do pipelienName not jobName and all other pre-carg nouns.

File: python_modules/dagster-graphql/dagster_graphql_tests/client_tests/test_submit_pipeline_execution.py (hunk 1 above)
Comment (alangenfeld) [reply python_modules/dagster-graphql/dagster_graphql_tests/client_tests/test_submit_pipeline_execution.py:86]: does get used in https://github.com/dagster-io/dagster/pull/6398?no-redirect=1

these tests are mock soup so the values are meaningless. I'll see if i can make anything meaningful happen or set up an integration test

Comment (yuhan) [reply python_modules/dagster-graphql/dagster_graphql_tests/client_tests/test_submit_pipeline_execution.py:86]: got it.

**Review by yuhan — APPROVED** [review]

**Pull request merged**